source/vir/src/ast_to_sst.rs

```diff
@@ -1462,13 +1462,10 @@ pub(crate) fn expr_to_stm_opt(
             let uid = state.declare_new_var(&tmp_ident, &expr.typ, false, false);
             let v = mk_exp(ExpX::Var(uid));
 
-            // assume v.len() == len
-            let len_eq = sst_equal(
-                &expr.span,
-                &sst_array_len(ctx, &expr.span, &v),
-                &sst_int_literal(&expr.span, elems.len() as i128),
-            );
-            stms.push(Spanned::new(expr.span.clone(), StmX::Assume(len_eq)));
+            // assume the type invariant
+            // (this implies that v.len() == len because of a vstd axiom)
```

Comment:
Which axiom?

Reply:
It's `array_len_matches_n` in pervasive/arrays.rs. I made the comment more specific.